src/main/scala/algolia/definitions/SynonymsDefinition.scala

```diff
@@ -174,7 +179,7 @@ case class BatchSynonymsDefinition(
     synonyms: Iterable[AbstractSynonym],
     index: Option[String] = None,
     forward: Option[ForwardToReplicas] = None,
-    replace: Option[ReplaceExistingSynonyms] = None,
+    replace: Option[ClearExistingSynonyms] = None,
```

Comment:
I have doubts about this being a BC!

Reply:
You're right I think that unfortunately it's a BC because Scala supports [named parameter](https://docs.scala-lang.org/tour/named-arguments.html) :( so if one user is using `ReplaceExistingSynonyms = ...` it will break their build.

So I am not sure if we should move forward with this PR in that case 🤔 What's your take on this @Aallam and @chloelbn?

Reply:
@Ant-hem you're saying there's no way to offer the possibility to pass both parameters and then only send `replaceExistingSynonyms` to the engine?